data-service/src/test/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryTestCase.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.repository;
+
+import com.amazonaws.blox.dataservice.mapper.EnvironmentMapper;
+import com.amazonaws.blox.dataservice.repository.model.EnvironmentDDBRecord;
+import com.amazonaws.blox.dataservice.repository.model.EnvironmentRevisionDDBRecord;
+import com.amazonaws.blox.dataservice.test.rules.LocalDynamoDb;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.model.CreateTableRequest;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput;
+import org.junit.Before;
+import org.junit.Rule;
+import org.mapstruct.factory.Mappers;
+
+public class EnvironmentRepositoryTestCase {
```

Comment:
`TestCase` -> `TestBase`?

Reply:
Hm. I've always named my test base classes `*TestCase` because of ye Olden Days where all JUnit tests had to derive from the `org.junit.TestCase` class, but I've not really thought about that in a long time. How about `AbstractEnvironmentRepositoryTest`?

Reply:
Which reminds me, this should be abstract...

Reply:
Y,  we have to make it as abstract if the test case class end with `Test`. Junit will fail to say no test case if a class ends with `Test` without actual test case.